test/e2e/json-rpc/switchEthereumChain.spec.js

```diff
@@ -158,7 +158,36 @@ describe('Switch Ethereum Chain for two dapps', function () {
         });
 
         await driver.switchToWindowWithUrl(DAPP_ONE_URL);
+        // Switch to Dapp One and connect it
+        await driver.switchToWindowWithUrl(DAPP_URL);
+        await driver.findClickableElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+        await driver.clickElement('#connectButton');
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        const editButtons = await driver.findElements('[data-testid="edit"]');
+
+        assert.ok(editButtons.length > 0, 'Edit buttons are available');
+
+        await editButtons[1].click();
```

Comment:
It looks like this would introduce a potential race condition (the edit buttons exist on line 170, but they may not be clickable, and they may not even exist anymore by this point). Typically I'd recommend doing the "find" and "interact" step in a single action, so that the element status doesn't change in-between the two actions. e.g. we can directly use `clickElement` here instead.

Reply:
Removed the assert statement here https://github.com/MetaMask/metamask-extension/pull/27980/commits/84afa0627172d542d511ea9c480f9993c9417291